zebrad/tests/acceptance.rs

```diff
@@ -1642,6 +1642,66 @@ async fn fully_synced_rpc_test() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn delete_old_databases() -> Result<()> {
+    use std::fs::create_dir;
+
+    zebra_test::init();
+
+    let mut config = default_test_config()?;
+    let run_dir = testdir()?;
+    let cache_dir = run_dir.path().join("state");
+
+    // create cache dir
+    create_dir(cache_dir.clone())?;
+
+    // create a v1 dir outside cache dir that should not be deleted
+    let outside_dir = run_dir.path().join("v1");
+    create_dir(&outside_dir)?;
+    assert!(outside_dir.as_path().exists());
+
+    // create a `v1` dir insidecache dir that should be deleted
+    let inside_dir = cache_dir.join("v1");
+    create_dir(&inside_dir)?;
+    assert!(inside_dir.as_path().exists());
+
+    // modify config with our cache dir and not ephimeral configuration
+    // (delete old databases function will not run when epehemeral = true)
+    config.state.cache_dir = cache_dir;
+    config.state.ephemeral = false;
+
+    // run zebra with our config
+    let mut child = run_dir
+        .with_config(&mut config)?
+        .spawn_child(args!["start"])?;
+
+    // delete checker running
+    child.expect_stdout_line_matches("checking for old database versions".to_string())?;
+
+    // inside dir was deleted
+    child.expect_stdout_line_matches(format!(
+        "deleted outdated state directory deleted_state={:?}",
```

Comment:
It looks like the new test never stops on macOS:
> test delete_old_databases has been running for over 60 seconds

https://github.com/ZcashFoundation/zebra/runs/6939346214?check_suite_focus=true#step:13:54440

It seems like the code runs correctly:
> zebra_state::config: deleted outdated state directory deleted_state="/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/zebrad_testsfBKR8x/state/v1"

https://github.com/ZcashFoundation/zebra/runs/6939346214?check_suite_focus=true#step:13:54350

But the regex in the test doesn't match.

This is probably because the code uses `canonicalize` on the path, but the test doesn't. (macOS `/tmp` is a symlink to its actual temporary directory.)